File: CPP/PROGRAM/Harddrive.cpp
#pragma once
#include <iostream>
#include <string>
#include "Komponen.cpp"
#include "StorageDevice.cpp"

using namespace std;

class Harddrive : public Komponen, public StorageDevice{
    //atribut
    private:
        string tipeDrive;

    public:
        //konstruktor
        Harddrive(){
            this->tipeDrive = "";
        }

        Harddrive(string tipeDrive, string merk, string nama, int kapasitasGB, float bacaTulisMBps) : Komponen(merk, nama), StorageDevice(kapasitasGB, bacaTulisMBps){
            this->tipeDrive = tipeDrive;
        }

        //getter dan setter
        void setTipeDrive(string tipeDrive){
            this->tipeDrive = tipeDrive;
        }
        
        string getTipeDrive(){
            return this->tipeDrive;
        }

        ~Harddrive(){

        }
};

File: PYTHON/PROGRAM/Harddrive.py
from Komponen import Komponen
from StorageDevice import StorageDevice

class Harddrive(Komponen, StorageDevice):
    #konstruktor
    def __init__(self, merk="", nama="", kapasitasGB=0, bacaTulisMBps=0, tipeDrive=""):
        Komponen.__init__(self, merk, nama)
        StorageDevice.__init__(self, kapasitasGB, bacaTulisMBps)
        self.__tipeDrive = tipeDrive

    #getter dan setter
    def getTipeDrive(self):
        return self.__tipeDrive
    
    def setTipeDrive(self, tipeDrive):
        self.__tipeDrive = tipeDrive